component/otelcol/connector/grafanacloud/connector.go

```diff
@@ -0,0 +1,120 @@
+package grafanacloud
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/connector"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+)
+
+const (
+	hostInfoMetric     = "host_info"
+	hostIdentifierAttr = "grafana.host.id"
+)
+
+var _ connector.Traces = (*connectorImp)(nil)
+
+type connectorImp struct {
+	config Config
+	logger *zap.Logger
+
+	started      bool
+	done         chan struct{}
+	shutdownOnce sync.Once
+
+	metricsConsumer consumer.Metrics
+	hostMetrics     *hostMetrics
+}
+
+func newConnector(logger *zap.Logger, config component.Config) *connectorImp {
+	logger.Info("Building grafanacloud connector")
+	cfg := config.(*Config)
+	return &connectorImp{
+		config:      *cfg,
+		logger:      logger,
+		done:        make(chan struct{}),
+		hostMetrics: newHostMetrics(),
+	}
+}
+
+// Capabilities implements connector.Traces.
+func (c *connectorImp) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+// ConsumeTraces implements connector.Traces.
+func (c *connectorImp) ConsumeTraces(ctx context.Context, td ptrace.Traces) error {
+	// loop through the levels of spans of the one trace consumed
```

Comment:
this comment is a leftover from the example connector in https://opentelemetry.io/docs/collector/building/connector/, we can keep it or remove it